v1-2/learn/how-to-use-java-interoperability.md

```diff
@@ -328,6 +328,155 @@ The next sections provide more details on various aspects related to Java intero
 
 ## The `bindgen` tool
 
+*Note that the bindgen tool is still experimental. We are in the process of improving the generated code.*
+
+The `bindgen` is a CLI tool that generates Ballerina bindings for Java classes.
+
+### The `bindgen` Command
+
+```sh
+ballerina bindgen [(-cp|--classpath) <classpath>...]
+                  [(-o|--output) <output>]
+                  (<class-name>...)
+```
+
+`(-cp|--classpath) <classpath>...`
+This optional parameter could be used to specify one or more comma-delimited classpaths for retrieving the required Java libraries needed by the bindgen tool execution. The classpath could be provided as comma separated paths of jar files or as comma separated paths of directories containing all the relevant Java libraries. If the Ballerina bindings are to be generated from a standard Java library or from a library available inside the Ballerina SDK, then you need not specify the classpath explicitly.
+
+`(-o|--output) <output>`
+This optional parameter could be used to specify the directory path into which the Ballerina bindings should be inserted. If this path is not specified, the output will be written onto the same directory from where the command is run. You can point the path of a Ballerina module to generate the code inside a Ballerina module. 
+
+`<class-name>...`
+One or more space separated fully qualified Java class names for which the Ballerina bridge code is to be generated. Please note that these class names should be provided at the end of the command.
+
+### Generated Bridge Code
+
+When the tool is run, a `.bal` file will be created to represent each Java class. This would contain the respective Ballerina object along with the required Java interoperability mappings. These `.bal` files would reside inside subdirectories representing the package structure. Apart from creating bindings for the Java classes specified, the command would also generate empty Ballerina objects for the dependent Java classes. A Java class would be considered dependent if it is used inside one of the Ballerina objects generated. A set of additional utility files will also be generated in order to support the auto generated Ballerina bindings. The folder structure of the generated bindings will be as follows.
```

Comment:
```suggestion
When the tool is run, a `.bal` file will be created to represent each Java class. This would contain the respective Ballerina object along with the required Java interoperability mappings. These `.bal` files would reside inside sub directories representing the package structure. Apart from creating bindings for the specified Java classes, the command would also generate empty Ballerina objects for the dependent Java classes. A Java class would be considered dependent if it is used inside one of the Ballerina objects generated. A set of additional utility files will also be generated in order to support the auto-generated Ballerina bindings. The folder structure of the generated bindings will be as follows.
```